nix/build-maven-package.nix

```diff
@@ -0,0 +1,74 @@
+{ lib, stdenv, maven }:
+
+{ src, sourceRoot ? null, buildOffline ? false, patches ? [ ], pname, version
+, mvnHash ? "", mvnFetchExtraArgs ? { }, mvnDepsParameters ? ""
+, manualMvnArtifacts ? [ ], mvnParameters ? "", ... }@args:
+
+# originally extracted from dbeaver
+# created to allow using maven packages in the same style as rust
+
+let
+  fetchedMavenDeps = stdenv.mkDerivation ({
+    name = "${pname}-${version}-maven-deps";
+    inherit src sourceRoot patches;
+
+    nativeBuildInputs = [ maven ];
+
+    buildPhase = ''
+      runHook preBuild
+    '' + lib.optionalString buildOffline ''
+      mvn org.apache.maven.plugins:maven-dependency-plugin:3.6.1:go-offline -Dmaven.repo.local=$out/.m2 ${mvnDepsParameters}
```

Comment:
using latest version of the go-offline plugin which does not build snapshot jars of the local project

Reply:
Um, you're not 100% requiring every single dependency to be pulled in by this, are you? go-offline has bugs that means it pulls in most of the dependencies, but not 100%. So if this is going to cause something to break if one or two dependencies isn't included in this, this solution isn't going to work.

Reply:
No, there are indeed dependencies that the `go-offline` plugin does not detect. The missed ones have to be manually included here:
https://github.com/runtimeverification/k/blob/fc1859b723b985582d4f02566c5be7cec1352916/flake.nix#L66-L71

I've added a section in the README troubleshooting explaining this:
https://github.com/runtimeverification/k/blob/fc1859b723b985582d4f02566c5be7cec1352916/README.md?plain=1#L435-L436